pkg/extensions/tbtc/tbtc.go

```diff
@@ -761,23 +790,58 @@ func (t *tbtc) watchKeepClosed(
 }
 
 func (t *tbtc) shouldMonitorDeposit(depositAddress string) bool {
```

Comment:
Can we add unit tests for this function covering caching? Just like we have for the firewall make sure that depending on `getSignerIndex` result, for the next `shouldMonitorDeposit` invocation, chain call is not executed and appropriate value is returned?